### ISSUE TITLE
[CT-644] Align Botocore and Boto3 Dependencies pinnings to support MWAA Airflow

### ISSUE THREAD

**Issue by FridayPush** [dbt-labs/dbt-redshift#118]
### Describe the feature
MWAA python package dependencies can not be resolved with the currently pinned ranges in `dbt-redshift`.

MWAA is a managed Airflow environment that runs in AWS. AWS Suggests a list of constraints to ensure that the environment is healthy when installing new packages. While I can imagine a desire to resolve conflicts with every platform the likelihood that an AWS environment is using Redshift seems pretty high. The constraint file for the most current version of MWAA can be found here:
https://raw.githubusercontent.com/apache/airflow/constraints-2.2.2/constraints-3.7.txt

It looks like MWAA wants boto3: to be `boto3==1.18.65`

For reference the logs when attempting to use `dbt-redshift` in the environment. 
```
ERROR: Cannot install dbt-redshift==0.13.0, dbt-redshift==0.13.1, dbt-redshift==0.14.0, dbt-redshift==0.14.1, dbt-redshift==0.14.2, dbt-redshift==0.14.3, dbt-redshift==0.14.4, dbt-redshift==0.15.0, dbt-redshift==0.15.1, dbt-redshift==0.15.2, dbt-redshift==0.15.3, dbt-redshift==0.16.0, dbt-redshift==0.16.1, dbt-redshift==0.17.0, dbt-redshift==0.17.1, dbt-redshift==0.17.2, dbt-redshift==0.18.0, dbt-redshift==0.18.1, dbt-redshift==0.18.2, dbt-redshift==0.19.0, dbt-redshift==0.19.1 and dbt-redshift==0.19.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    dbt-redshift 0.19.2 depends on boto3<1.16 and >=1.4.4
    dbt-redshift 0.19.1 depends on boto3<1.16 and >=1.4.4
    dbt-redshift 0.19.0 depends on boto3<1.16 and >=1.4.4
    dbt-redshift 0.18.2 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.18.1 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.18.0 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.17.2 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.17.1 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.17.0 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.16.1 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.16.0 depends on botocore<1.15 and >=1.5.0
    dbt-redshift 0.15.3 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.15.2 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.15.1 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.15.0 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.14.4 depends on boto3<1.11.0 and >=1.4.4
    dbt-redshift 0.14.3 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.14.2 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.14.1 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.14.0 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.13.1 depends on boto3<1.10.0 and >=1.6.23
    dbt-redshift 0.13.0 depends on boto3<1.10.0 and >=1.6.23
    The user requested (constraint) boto3==1.18.65
    The user requested (constraint) botocore==1.21.65
```

### Describe alternatives you've considered
We'll continue to use the 'dbt' package and call outs to the external cli.

### Who will this benefit?
Users that run DBT via Airflow.

### Are you interested in contributing this feature?
I can help look into it, but I'm unfamiliar with the historical reason the packages are pinned as they are.

**Comment by jtcohen6** [comment]
@FridayPush It looks like you're trying to install older versions of `dbt-redshift`. The latest version of `dbt-redshift` has an upper bound on `boto3<2.0.0`, which should be compatible with MWAA's constraints:

https://github.com/dbt-labs/dbt-redshift/blob/50399f56d912c30a55d813ec61047fb3dc9da223/setup.py#L71-L72

That was bumped in https://github.com/dbt-labs/dbt-core/pull/3148 (March 2021), and included in all `dbt-redshift` versions since v0.20.

**Comment by Fleid** [comment]
Closing as stale. Let us know if this should be re-opened.
